### PR TITLE
Automate smoke tests for every feature

### PR DESCRIPTION
**Pull request by PatrickF1**
While I can't and don't plan on trying to test every possible code path, I can at least set up smoke tests. These smoke tests should catch many cases where a regression would completely break a feature, and thus prevent regressions from getting merged in. So while it won't obviate manual testing, it will cut down on manual testing for both me and anyone who wants to contribute to fzf.fish!

### CODE REVIEW

File: tests/search_current_dir/expands_~_base_dir.fish [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# mock everything so that the only thing that gets sent to stdout are the args passed into fd
+mock fd \* "echo \$argv"
+mock fzf \* ""
+mock commandline --current-token "echo ~/"
+mock commandline \* ""
+set fd_args (__fzf_search_current_dir)
+@test "~/ not in fd args" -z (string match "~/" $fd_args)
+@test "~/ is expanded to $HOME" -n (string match --entire "$HOME" $fd_args)
```
Comment (jorgebucaran) [inline tests/search_current_dir/expands_~_base_dir.fish:8]: As a coding convention, I always specify `--`:
```fish
string match --entire -- "$HOME" $fd_args
```

**Review by jorgebucaran — APPROVED** [review]

File: tests/search_current_dir/expands_~_base_dir.fish (hunk 1 above)
Comment (PatrickF1) [reply tests/search_current_dir/expands_~_base_dir.fish:8]: Thanks Jorge! That's an important convention I will copy.

**Comment by PatrickF1** [comment]
Thanks for approving. I'm actually gonna add a lot more tests. Would appreciate another review later 😉

File: functions/__fzf_preview_file.fish [hunk 2]
```diff
@@ -23,6 +23,6 @@ function __fzf_preview_file --argument-names file_path --description "Print a pr
     else if test -p "$file_path"
         __fzf_report_file_type "$file_path" "named pipe"
     else
-        echo "Unknown file type." >&2
+        echo "File doesn't exist." >&2
```
Comment (PatrickF1) [inline functions/__fzf_preview_file.fish:26]: Fixed a small bug: realized the only time this happens is if the file doesn't exist.

**Pull request merged**